doc/tutorials/klondike/app/lib/step5/klondike_game.dart

```diff
@@ -0,0 +1,294 @@
+              nMovingCards--;
+              if (nMovingCards == 0) {
+                deal(startType);
+              }
+            },
+          );
+        }
+      }
+    }
+  }
+
+  void deal(Startup startType) {
+    switch (startType) {
+      case Startup.first:
+      case Startup.newDeal:
+        cards.shuffle();
+        break;
+      case Startup.changeDraw:
+        _klondikeDraw = (_klondikeDraw == 3) ? 1 : 3;
+        cards.shuffle();
+        break;
+      case Startup.sameDeal:
+      default:
+        break;
+    }
+
+    var cardToDeal = cards.length - 1;
+    var nMovingCards = 0;
+    for (var i = 0; i < 7; i++) {
+      for (var j = i; j < 7; j++) {
+        final card = cards[cardToDeal--];
+        card.doMove(
+          piles[j].position,
+          start: nMovingCards * 0.15,
+          onComplete: () {
+            piles[j].acquireCard(card);
+            nMovingCards--;
+            if (nMovingCards == 0) {
+              var delayFactor = 0;
+              for (final pile in piles) {
+                delayFactor++;
+                pile.flipTopCard(start: delayFactor * 0.15);
+              }
+            }
+          },
+        );
+        nMovingCards++;
+      }
+    }
+    for (var n = 0; n <= cardToDeal; n++) {
+      stock.acquireCard(cards[n]);
+    }
+  }
+
+  void addButton(String label, double buttonX, Startup action) {
+    final button = FlatButton(
+      label,
+      size: Vector2(cardWidth, 0.6 * topGap),
+      position: Vector2(buttonX, topGap / 2),
+      onReleased: () {
+        if (action == Startup.haveFun) {
+          letsCelebrate();
+        } else {
+          init(action);
+        }
+      },
+    );
+    world.add(button);
+  }
+
+  void checkWin() {
+    var nComplete = 0;
+    for (final f in foundations) {
+      if (f.isFull) {
+        nComplete++;
+      }
+    }
+    if (nComplete == foundations.length) {
+      letsCelebrate();
+    }
+  }
+
+  void letsCelebrate({int phase = 1}) {
+    // Deal won: bring all cards to the middle of the screen (phase 1)
+    // then scatter them to points just outside the screen (phase 2).
+    //
+    // First get the device's screen-size in game co-ordinates, then get the
+    // top-left of the off-screen area that will accept the scattered cards.
+    // Note: The play area is anchored at TopCenter, so topLeft.y is fixed.
+
+    final cameraZoom = camera.viewfinder.zoom;
+    final zoomedScreen = size / cameraZoom;
+    final playArea =
+        Vector2(7 * cardSpaceWidth + cardGap, 4 * cardSpaceHeight + topGap);
+    final screenCenter = (playArea - cardSize) / 2;
+    final topLeft = Vector2(
+      (playArea.x - zoomedScreen.x) / 2 - cardWidth,
+      -cardHeight,
+    );
+    final nCards = cards.length;
+    final h = zoomedScreen.y + cardSize.y; // Height of offscreen rect.
+    final w = zoomedScreen.x + cardSize.x; // Width of offscreen rect.
+    final ds = (h + w + h + w) / nCards; // Spacing = perimeter / nCards.
```

Comment:
Rename to not use abbreviations for better readability

Reply:
Done.